contributed_definitions/nyaml/NXinstrument.yaml

```diff
@@ -0,0 +1,190 @@
+category: base
+doc: |
+  Collection of the components of the instrument or beamline.
+  
+  Template of instrument descriptions comprising various beamline components.
+  Each component will also be a NeXus group defined by its distance from the
+  sample. Negative distances represent beamline components that are before the
+  sample while positive distances represent components that are after the sample.
+  This device allows the unique identification of beamline components in a way
+  that is valid for both reactor and pulsed instrumentation.
+type: group
+NXinstrument(NXobject):
+  name:
+    doc: |
+      Name of instrument
+    \@short_name:
+      doc: |
+        short name for instrument, perhaps the acronym
+  energy_resolution(NX_FLOAT):
+    unit: NX_ENERGY
+    doc: |
+      Energy resolution of the experiment (FWHM or gaussian broadening)
+  momentum_resolution(NX_FLOAT):
+    unit: NX_WAVENUMBER
+    doc: |
+      Momentum resolution of the experiment (FWHM)
+  angular_resolution(NX_FLOAT):
+    unit: NX_ANGLE
+    doc: |
+      Angular resolution of the experiment (FWHM)
+  spatial_resolution(NX_FLOAT):
+    unit: NX_LENGTH
+    doc: |
+      Spatial resolution of the experiment (Airy disk radius)
+  temporal_resolution(NX_FLOAT):
+    unit: NX_TIME
+    doc: |
+      Temporal resolution of the experiment (FWHM)
```

Comment:
Use of one generic NXresolution base class here

Reply:
More specific resolutions defined in the AppDef

Reply:
So you don't want to have these specific resolutions on the base class level of `NXinstrument`. We have them in `NXelectronanalyser` as well.

Reply:
I'm not 100% sure what is the best way here, maybe we can discuss this today in the TF meeting. Any listing will most likely be incomplete, that's why this flexible solution with a generic base class has its charm. On the other hand, one should define names for the concepts we can define now.

Reply:
My idea was to have both, i.e., some define names for concepts that are common (energy, momentum, time, spatial, angular resolution) and then also have a generic `(NXresolution)` for other types of resolution we don't know about now. See [NXinstrument](https://github.com/FAIRmat-NFDI/nexus_definitions/blob/a74768b36da66b0f68202c16a71d54e9664941b2/contributed_definitions/nyaml/NXinstrument.yaml) in #124.